src/python/pants/core/util_rules/external_tool.py

```diff
@@ -175,6 +224,35 @@ def get_request(self, plat: Platform) -> ExternalToolRequest:
             f"{self.known_versions}"
         )
 
+    @classmethod
+    def check_version_constraints(
+        cls, version: str, constraints: Requirement, action: UnsupportedVersionUsage
+    ) -> None:
+        if constraints.specifier.contains(version):  # type: ignore[attr-defined]
+            # all ok
+            return None
+
+        msg = [
+            f"The option [{cls.options_scope}].version is set to {version}, which is not compatible with what this release of Pants expects: {constraints}",
+        ]
+
+        if action is UnsupportedVersionUsage.Allow:
+            msg.append(
+                f"The option [{cls.options_scope}].use_unsupported_version is set to 'allow'."
+            )
+            logger.info("\n".join(msg))
+        else:
+            msg.append(
+                "Please update the version to a supported value, or consider using a different Pants release if you cannot change the version."
+            )
+            if action is UnsupportedVersionUsage.LogWarning:
+                logger.warning("\n".join(msg))
```

Comment:
```suggestion
                logger.warning("\n\n".join(msg))
```

Reply:
I had two newlines first, but thought it would use to much space, as it is for the logs, not regular console output. But if that's prefered any way, I can add it back :)

Reply:
In that case, maybe join with " " so that it's all one paragraph? I personally think a single newline ends up not looking very good in CLIs, so I always use two new lines or none at all.

Reply:
Yeah, it'll be a rather long line, but with word wrapping in the terminal/reader, that shouldn't be an issue. I found the lines to be tighter connected than to have them all on separate "paragraphs".